### PR TITLE
Added option to adjust resource dimming style / Refactored resource dimming logic

### PR DESCRIPTION
**Pull request by zeyangl**
* added a resource icon option in cqui settings. It has three options
- 0: always resource icons.
- 1: draw optimally improved resources as transparent.
- 2: hide optimally improved resources.
* fixed an lua exception when drawing archaeology site icon. archaeology site does not have an improvement table, but the old code was accessing the first improvement nonetheless. This bug luckily caused not visual bug, only a bunch of exception logs in the game log.

### CODE REVIEW

**Comment by chaorace** [comment]
In the future, please use pull --rebase to avoid creating a merge commit. Creating merge commits is only acceptable for major revisions and should not be applied in the case of a minor catchup merge. Don't sweat it too much since it looks like I have write access to your PR and can just rewrite the change history, but it's something to consider in the future for making both our lives easier!

File: CQUI/CQUI_Settings.sql [hunk 1]
```diff
@@ -36,4 +36,5 @@ INSERT INTO CQUI_Settings -- Don't touch this line!
 */
 
 INSERT INTO CQUI_Settings -- Don't touch this line!
-  VALUES  ("CQUI_BindingsMode", 1); -- Set of keybindings used │ 0=Civ6 default │ 1=keybinds from Civ5 │ 2=Civ5 with additions such as WASD camera control │
```
Comment (chaorace) [inline CQUI/CQUI_Settings.sql]: Values should be assigned in alphabetical order

File: CQUI/CQUI_SettingsElement.lua [hunk 2]
```diff
@@ -290,6 +297,8 @@ function Initialize()
   --Populating/binding comboboxes...
   --PopulateComboBox(Controls.BindingsPullDown, bindings_options, "CQUI_BindingsMode", Locale.Lookup("LOC_CQUI_BINDINGS_DROPDOWN_TOOLTIP"));
   PopulateComboBox(Controls.BindingsPullDown, bindings_options, "CQUI_BindingsMode", "Standard: Unchanged[NEWLINE]Classic: Civ V binds[NEWLINE]Enhanced: Civ V Binds with the following changes[NEWLINE]  WASD camera control[NEWLINE]  Q/E unit/city cycling[NEWLINE]  Shift toggles city/unit selection[NEWLINE]  Quarry/Airstrike are moved to alt-key + Q/S[NEWLINE]  NOTE:UNBIND W/E IN SETTINGS OR THINGS WON'T WORK!");
+
```
Comment (chaorace) [inline CQUI/CQUI_SettingsElement.lua]: This needs to have a localized counterpart ready for use when firaxis fixes custom LOC

File: CQUI/CQUI_Settings.sql [hunk 3]
```diff
@@ -37,7 +37,8 @@ INSERT INTO CQUI_Settings -- Don't touch this line!
...
```
Comment (chaorace) [inline CQUI/CQUI_Settings.sql]: Values should be assigned in alphabetical order

File: UI/WorldViewIconsManager.lua [hunk 4]
```diff
@@ -29,6 +29,8 @@ local m_RecommendedImprovementPlots :table = {};
 -- Used to efficiently clear settlement recommendations
 local m_RecommendedSettlementPlots :table = {};
 
+local g_resource_icon_style = 1;
```
Comment (chaorace) [inline UI/WorldViewIconsManager.lua]: This value should be prefixed with "CQUI_" instead of "g_"

File: CQUI/CQUI_Settings.sql [hunk 5]
```diff
@@ -37,7 +37,8 @@ INSERT INTO CQUI_Settings -- Don't touch this line!
...
+  VALUES  ("CQUI_BindingsMode", 1), -- Set of keybindings used │ 0=Civ6 default │ 1=keybinds from Civ5 │ 2=Civ5 with additions such as WASD camera control │
+      ("CQUI_ResourceIconStyle", 0);
```
Comment (chaorace) [inline CQUI/CQUI_Settings.sql]: This needs to have a comment concerning the meaning of the property

**Comment by zeyangl** [comment]
I will definitely take note of the issues in your comments. It will take me a couple of days to get around and make the changes though. So feel free to make the changes.

**Pull request merged**